### PR TITLE
Add intersphinx config, docs on how to use

### PR DESCRIPTION
**Pull request by wild-endeavor**
See docs

* Ran pycharm format
* Removed an unused Make target, I think.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,11 +20,6 @@ end2end:
 end2end_execute:
 	@end2end/execute.sh
 
-# Use this target to build the rsts directory only. In order to build the entire flyte docs, use update_ref_docs && all_docs
-.PHONY: generate-local-docs
-generate-local-docs:
-	@docker run -t -v `pwd`:/base ghcr.io/nuclyde-io/docbuilder:e70cfafe3397c3b23d11183973c0f44e0024f025 sphinx-build -E -b html /base/rsts/. /base/_build
-
 # Builds the entire doc tree. Assumes update_ref_docs has run and that all externals rsts are in _rsts/ dir
 .PHONY: generate-docs
```
Comment (kumare3) [inline Makefile:24]: remove this one too

**Comment by kumare3** [comment]
Small change

File: rsts/conf.py [hunk 2]
```diff
@@ -201,13 +194,24 @@
...
+# intersphinx configuration
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3", None),
+    "numpy": ("https://numpy.org/doc/stable", None),
+    "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
+    "torch": ("https://pytorch.org/docs/master/", None),
+    "scipy": ("https://docs.scipy.org/doc/scipy/reference", None),
+    "matplotlib": ("https://matplotlib.org", None),
+    "flytekit": ("https://flyte.readthedocs.io/projects/flytekit/en/master/", None),
+    # "flytekit": ("/Users/ytong/go/src/github.com/flyteorg/flytekit/docs/build/html", None),
```
Comment (katrogan) [inline rsts/conf.py:211]: add a note to uncomment for local development, etc?

File: rsts/contributor/docs/index.rst [hunk 3]
```diff
@@ -38,8 +40,128 @@ Building all documentation can be a slow process. To speed up the iteration one
...
+    Attempting "https://flytekit.readthedocs.io/en/latest/objects.inv" ...
+    Remote inventory found.
+
+    :py:function:`flytekit.task`
+    :std:doc:`tasks`
+    :std:doc:`tasks.extend`
+    :std:label:`tasks:tasks`
+
+    # Using the CLI to query a local file, useful when iterating locally
+    $ sphobjinv suggest ~/go/src/github.com/flyteorg/flytekit/docs/build/html/objects.inv task
```
Comment (katrogan) [inline rsts/contributor/docs/index.rst:99]: cool!

**Review by katrogan — DISMISSED** [review]
ty for this meticulous write-up!

File: Makefile (hunk 1 above)
Comment (wild-endeavor) [reply Makefile]: done.

**Review by katrogan — APPROVED** [review]

**Pull request merged**